OperatorOverloading/01 Operator Overloading/ClassLibrary1/FileExchangeRateProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OperatorOverloading.Parse;
+using System.Configuration;
+
+namespace OperatorOverloading.Dbl
+{
+    public class FileExchangeRateProvider:IExchangeRateProvider
+    {
+         static string[] initialseperatedString;
+         static string[] currencySeperatedString;
+
+        public  double GetExchangeRate(string sourceCurrency,string targetCurrency)
+        {
+            string line;
+            string completeData = "";
+
+            System.IO.StreamReader file = new System.IO.StreamReader(ConfigurationManager.AppSettings["Path"]);
+            while ((line = file.ReadLine()) != null)
+            {
+                completeData += line;
+            }
+            file.Close();
+            initialseperatedString = completeData.Split('{');
+            initialseperatedString[0] = "";
+            initialseperatedString[1] = "";
+
+            currencySeperatedString = initialseperatedString[2].Split(',');
+
+            double converter1 = GetConverter(sourceCurrency);
+            double converter2 = GetConverter(targetCurrency);
+
+            return (converter2/converter1);
+        }
+        public static double GetConverter(string sourceCurrency)
+         {
```

Comment:
Read about exchange rates. Just because USD -> INR = 63 and INR -> GBP = 0.01
does not mean USD -> GBP = 0.63. It is actually 0.64.